File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/EditTextViewHolder.java
package com.supercilex.robotscouter.ui.scout.viewholder;

import android.support.annotation.CallSuper;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.supercilex.robotscouter.R;

public class EditTextViewHolder extends ScoutViewHolderBase<String, TextView>
        implements View.OnFocusChangeListener {
    private final EditText mNotes;

    public EditTextViewHolder(View itemView) {
        super(itemView);
        mNotes = (EditText) itemView.findViewById(R.id.notes);
    }

    @Override
    @CallSuper
    public void bind() {
        super.bind();
        mNotes.setText(mMetric.getValue());
        mNotes.setOnFocusChangeListener(this);
    }

    @Override
    @CallSuper
    public void onFocusChange(View v, boolean hasFocus) {
        if (!hasFocus && v.getId() == R.id.notes) {
            updateMetricValue(mNotes.getText().toString());
        }
    }
}

File: app/src/main/java/com/supercilex/robotscouter/ui/scout/viewholder/EditTextViewHolder.kt
package com.supercilex.robotscouter.ui.scout.viewholder

import android.support.annotation.CallSuper
import android.support.design.widget.TextInputLayout
import android.view.View
import android.widget.TextView
import com.supercilex.robotscouter.R
import com.supercilex.robotscouter.data.model.Metric

open class EditTextViewHolder(itemView: View) :
        ScoutViewHolderBase<Metric.Text, String?, TextView>(itemView),
        View.OnFocusChangeListener {
    private val textLayout: TextInputLayout = itemView.findViewById(R.id.text_layout)

    public override fun bind() {
        super.bind()
        name.text = metric.value
        textLayout.hint = metric.name
        name.onFocusChangeListener = this
    }

    @CallSuper
    override fun onFocusChange(v: View, hasFocus: Boolean) {
        if (!hasFocus) updateMetricValue(name.text.toString())
    }
}
